Bulldozer.CSV/Maps/Communication.cs

```diff
@@ -0,0 +1,233 @@
+                    FromEmail = communicationCsv.FromEmail,
+                    ReplyToEmail = communicationCsv.ReplyToEmail,
+                    BCCEmails = communicationCsv.BCCEmails,
+                    Subject = communicationCsv.Subject,
+                    Message = communicationCsv.EmailMessage,
+                    SMSMessage = communicationCsv.SMSMessage,
+                    CommunicationType = ( CommunicationType ) ( int ) communicationCsv.CommunicationType,
+                    SenderPersonAliasId = sender != null ? sender.PersonAliasId : ImportPersonAliasId,
+                    SendDateTime = communicationCsv.SentDateTime.GetValueOrDefault() < RockDateTime.Now ? communicationCsv.SentDateTime.GetValueOrDefault() : new DateTime( 1900, 1, 1 ), 
+                    CreatedByPersonAliasId = createdByPerson != null ? createdByPerson.PersonAliasId : ImportPersonAliasId,
+                    ForeignId = communicationCsv.CommunicationId.AsIntegerOrNull(),
+                    ForeignKey = $"{this.ImportInstanceFKPrefix}^{communicationCsv.CommunicationId}",
+                    CreatedDateTime = communicationCsv.DateCreated, 
+                    Status = CommunicationStatus.Approved
+                };
+                newCommunications.Add( newCommunication );
+            }
+
+            rockContext.BulkInsert( newCommunications );
+            return communicationCsvList.Count;
+        }
+
+        /// <summary>
+        /// Loads the CommunicationRecipient data.
+        /// </summary>
+        private int LoadCommunicationRecipient()
+        {
+            this.ReportProgress( 0, "Preparing CommunicationRecipient data for import" );
+
+            var rockContext = new RockContext();
+            var importedCommunications = rockContext.Communications.AsNoTracking()
+                .Where( c => c.ForeignKey != null && c.ForeignKey.StartsWith( this.ImportInstanceFKPrefix + "^" ) )
+                .ToList()
+                .ToDictionary( k => k.ForeignKey, v => v );
+
+            var importedCommunicationRecipients = rockContext.CommunicationRecipients.AsNoTracking()
+                .Where( c => c.ForeignKey != null && c.ForeignKey.StartsWith( this.ImportInstanceFKPrefix + "^" ) )
+                .ToList()
+                .ToDictionary( k => k.ForeignKey, v => v );
+
+            var communicationRecipientCsvsToProcess = this.CommunicationRecipientCsvList.Where( c => !importedCommunicationRecipients.ContainsKey( $"{this.ImportInstanceFKPrefix}^{c.CommunicationRecipientId}" ) && importedCommunications.ContainsKey( $"{this.ImportInstanceFKPrefix}^{c.CommunicationId}" ) );
+
+            if ( communicationRecipientCsvsToProcess.Count() < this.CommunicationRecipientCsvList.Count() )
+            {
+                this.ReportProgress( 0, $"{this.CommunicationRecipientCsvList.Count() - communicationRecipientCsvsToProcess.Count()} CommunicationRecipients(s) from import already exist and will be skipped." );
```

Comment:
```suggestion
                this.ReportProgress( 0, $"{this.CommunicationRecipientCsvList.Count() - communicationRecipientCsvsToProcess.Count()} CommunicationRecipients(s) have been skipped because they already exist or their communication id does not exist." );
```